app/src/main/kotlin/com/hippo/ehviewer/download/DownloadManager.kt

```diff
@@ -530,9 +538,12 @@ object DownloadManager : OnSpiderListener, CoroutineScope {
     }
 
     suspend fun readMetadataFromLocal() {
-        val list = allInfoList.filter { it.pages == 0 || it.simpleTags == null }.parMapNotNull(concurrency = 5) { info ->
+        val mutex = Mutex()
+        val cache = mutableMapOf<DownloadInfo, ComicInfo?>()
+
+        val galleryInfoList = allInfoList.filter { it.pages == 0 || it.simpleTags == null }.parMapNotNull(concurrency = 5) { info ->
             info.downloadDir?.run {
-                val comicInfo = findFile(COMIC_INFO_FILE)?.let { readComicInfo(it) }
+                val comicInfo = findFile(COMIC_INFO_FILE)?.let { readComicInfo(it) }?.also { mutex.withLock { cache[info] = it } }
                 if (comicInfo != null) {
                     info.pages = comicInfo.pageCount
                     info.simpleTags = comicInfo.toSimpleTags()
```

Comment:
Could just add DownloadArtist to the pending list instead of reading ComicInfo twice.